### PR TITLE
Fixed webhook and support multiple instances of the operator in the same cluster

### PR DESCRIPTION
**Pull request by yuchaoran2011**
This PR has the following contents:

1. Fixed the webhook together with this [PR](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/342).
2. Made names of cluster-wide/non-namespaced resources such as ClusterRole, ClusterRoleBinding and MutatingWebhookConfiguration unique for each deployment. Previously, because these names did not vary for each Helm deployment, each cluster could only have one operator instance.
3. Combined spark job namespace and the namespace(s) that the operator monitors into a single parameter called `sparkJobNamespace`.

@liyinan926 I temporarily changed the image in `values.yaml` to use my test image. Once you rebuild and push `gcr.io/spark-operator/spark-operator:v2.4.0-v1alpha1-latest` with this [PR](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/342), I'll change it back. But unfortunately the CI is still failing with the "`spark-webhook-secrets` not found" error. I'm confused how my changes can cause it to happen. I'm not able to reproduce it on my cluster.

Do you think we want to set the webhook to be disabled by default in the chart in order to get this PR in?

### CODE REVIEW

File: incubator/sparkoperator/values.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
-operatorImageName: gcr.io/spark-operator/spark-operator
-operatorVersion: v2.3.1-v1alpha1-latest
-sparkJobNamespace: default
+operatorImageName: lightbend/sparkoperator
```
Comment (liyinan926) [inline incubator/sparkoperator/values.yaml]: Please use `gcr.io/spark-operator/spark-operator`.

**Comment by liyinan926** [comment]
@yuchaoran2011 I'm fine with. We also need to update the documentation in the operator repo to make it clear.

**Comment by yuchaoran2011** [comment]
@liyinan926 Sure sounds good. I'll do that once this one is merged.

**Comment by yuchaoran2011** [comment]
@liyinan926 The CI is now passing. Needs your lgtm and it's good to be merged. I'll update the operator doc tomorrow.

**Pull request merged**